### PR TITLE
site: hide theme example menu's expandIcon

### PR DESCRIPTION
**Pull request by kiner-tang**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [X] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  -  |
| 🇨🇳 Chinese | -  |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [X] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [X] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 35d595a</samp>

This pull request enhances the `Menu` component by adding a new prop type, a new test case, and a more flexible way to customize the expand icon. It also updates `rc-menu` and removes an unused dependency from `package.json`.

### <samp>🤖 Generated by Copilot at 35d595a</samp>

*  Update `rc-menu` dependency to version `9.12.0` for bug fixes and improvements ([link](https://github.com/ant-design/ant-design/pull/44552/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L137-R137))
*  Import `ReactNode` type from `react` and use it as prop type for `expandIcon` prop of `Menu` component ([link](https://github.com/ant-design/ant-design/pull/44552/files?diff=unified&w=0#diff-1cad196e610d68ddf06e127c070d41cb9e4bfa00016e77c472841f261f154433L8-R8), [link](https://github.com/ant-design/ant-design/pull/44552/files?diff=unified&w=0#diff-bcc9a016f0b5e2844ec9d0aa399d06a86134f535931eeced33c3d08c3462ce43R20))
*  Merge default and custom props for `Menu` component using `lodash.merge` and handle cases when `expandIcon` is `null` or `false` to hide expand icon for submenus ([link](https://github.com/ant-design/ant-design/pull/44552/files?diff=unified&w=0#diff-bcc9a016f0b5e2844ec9d0aa399d06a86134f535931eeced33c3d08c3462ce43L129-R138))
*  Remove unnecessary type assertions for `menuProps` object in `useMemo` hook in `components/menu/__tests__/index.test.tsx` ([link](https://github.com/ant-design/ant-design/pull/44552/files?diff=unified&w=0#diff-1cad196e610d68ddf06e127c070d41cb9e4bfa00016e77c472841f261f154433L828-R828), [link](https://github.com/ant-design/ant-design/pull/44552/files?diff=unified&w=0#diff-1cad196e610d68ddf06e127c070d41cb9e4bfa00016e77c472841f261f154433L863-R863))
*  Add test case for `Menu` component with `expandIcon` set to `null` or `false` in `components/menu/__tests__/index.test.tsx` ([link](https://github.com/ant-design/ant-design/pull/44552/files?diff=unified&w=0#diff-1cad196e610d68ddf06e127c070d41cb9e4bfa00016e77c472841f261f154433R1126-R1156))
*  Add and remove `@biomejs/biome` dependency in `package.json`, probably by mistake or test ([link](https://github.com/ant-design/ant-design/pull/44552/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R168), [link](https://github.com/ant-design/ant-design/pull/44552/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L294))

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -134,7 +134,7 @@
     "rc-input": "~1.1.0",
     "rc-input-number": "~8.0.2",
     "rc-mentions": "~2.5.0",
-    "rc-menu": "~9.10.0",
+    "rc-menu": "~9.12.0",
```
[the remark this answers is not in the document]
Comment (kiner-tang) [reply package.json:137]: 好的
